core/runtime/src/main/java/io/quarkus/runtime/graal/CidrAddressSubstitutions.java

```diff
@@ -0,0 +1,93 @@
+
+/*
+ * The following substitutions are required because of a new restriction in GraalVM 19.3.0 that prohibits the presence of
+ * java.net.Inet4Address and java.net.Inet6Address in the image heap. Each field annotated with @InjectAccessors is lazily
+ * recomputed at runtime on first access while CidrAddress.class can still be initialized during the native image build.
+ */
+@TargetClass(CidrAddress.class)
+final class Target_org_wildfly_common_net_CidrAddress {
+
+    @Alias
+    @InjectAccessors(Inet4AnyCidrAccessor.class)
+    public static CidrAddress INET4_ANY_CIDR;
+
+    @Alias
+    @InjectAccessors(Inet6AnyCidrAccessor.class)
+    public static CidrAddress INET6_ANY_CIDR;
+}
+
+class Inet4AnyCidrAccessor {
+
+    private static volatile CidrAddress INET4_ANY_CIDR;
+
+    static CidrAddress get() {
+        CidrAddress result = INET4_ANY_CIDR;
+        if (result == null) {
+            // Lazy initialization on first access.
+            result = initializeOnce();
+        }
+        return result;
+    }
+
+    private static synchronized CidrAddress initializeOnce() {
+        CidrAddress result = INET4_ANY_CIDR;
+        if (result != null) {
+            // Double-checked locking is OK because INSTANCE is volatile.
+            return result;
+        }
+        try {
+            Constructor<CidrAddress> ctor = CidrAddress.class.getDeclaredConstructor(InetAddress.class, int.class);
+            ctor.setAccessible(true);
+            result = ctor.newInstance(Inet.INET4_ANY, 0);
+        } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException
+                | IllegalArgumentException | InvocationTargetException e) {
+            throw new RuntimeException("CidrAddress instanciation failed", e);
+        }
+        INET4_ANY_CIDR = result;
+        return result;
+    }
+}
+
+class Inet6AnyCidrAccessor {
+
+    private static volatile CidrAddress INET6_ANY_CIDR;
+
+    static CidrAddress get() {
+        CidrAddress result = INET6_ANY_CIDR;
+        if (result == null) {
+            // Lazy initialization on first access.
+            result = initializeOnce();
+        }
+        return result;
+    }
+
+    private static synchronized CidrAddress initializeOnce() {
+        CidrAddress result = INET6_ANY_CIDR;
+        if (result != null) {
+            // Double-checked locking is OK because INSTANCE is volatile.
+            return result;
+        }
+        try {
+            Constructor<CidrAddress> ctor = CidrAddress.class.getDeclaredConstructor(InetAddress.class, int.class);
+            ctor.setAccessible(true);
+            result = ctor.newInstance(Inet.INET6_ANY, 0);
+        } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException
+                | IllegalArgumentException | InvocationTargetException e) {
+            throw new RuntimeException("CidrAddress instanciation failed", e);
+        }
```

Comment:
@dmlloyd Is this acceptable? The `CidrAddress` constructor is `private` so I need a way to access it.

Reply:
Use an alias for the constructor instead.

Reply:
I tried the alias way last night and it didn't work, but it was super late for me so I didn't really try to analyze the reason and decided to test this alternative that I knew would work. I'll replace this with an alias again tonight and see if I can make it work. Thanks.